app/data_models/progress_store.py

```diff
@@ -76,87 +77,93 @@ def is_dirty(self) -> bool:
     def is_routing_backwards(self) -> bool:
         return self._is_routing_backwards
 
-    def is_section_complete(
+    def is_section_or_list_item_complete(
```

Comment:
Not sure about some of the renaming being done in this class e.g. `is_section_or_list_item_complete`, can a list item be in progress/complete? We are really only checking completeness of sections and not list_items just that some sections have attached list_item_ids when in repeats?

Reply:
After re-reveiwing, let me know if methods should be renamed `is_section_or_list_item_complete` to `is_progress_complete` - same for other similar method names.

Reply:
I think I'd be in favour of the method name sticking with what its checking
Maybe it could be something like `is_section_or_list_item_repeating_blocks_complete`? but im not sure if thats just more convoluted... in which case perhaps progress is the name to go for

Reply:
I'll go with `is_progress-complete` and rely on the new docstrings to fully explain what is being elements are having their progresses tracked.

Reply:
Renamed to use `progress` as the term